tests/tests/test_resource_limits.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3, as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import subprocess
+import time
+
+import Qemu
+import util
+
+def test_huge_resource_vm():
+    """
+    Test huge resources  (Intel Case ID 007)
+    """
+
+    # choose half available memory per Intel case spec
+    huge_mem_gb = int(util.get_memory_free_gb() / 2)
```

Comment:
here we are creating VM with half of free memory
```
*               total        used        free      shared  buff/cache   available
Mem:            13Gi       6.6Gi       1.5Gi       211Mi       5.9Gi       6.9Gi
Swap:          4.0Gi       1.0Mi       4.0Gi
```
for this example, the free memory is `1.5Gb`.
We are following the instructions at : https://github.com/intel/tdx/wiki/Tests
however, i believe we could do better, at least we can use the `available` or even `total` value

Reply:
I think if we us total or available we could end up asking for more than is available to use.  It seems we should use either half of free or some other percentage.  We can always add another one that adds 75% of free or 90% of free or even 99% of free....

Reply:
for `total`, i have to check how qemu asks for memory (let me do a quick check), but for `available`, i think that should be fine (https://askubuntu.com/questions/867068/what-is-available-memory-while-using-free-command).

Reply:
https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/7/html/virtualization_deployment_and_administration_guide/chap-overcommitting_with_kvm

Reply:
Just did the test, i can start a VM with 125G RAM when my host has only less than 70G available

![image](https://github.com/user-attachments/assets/a1570548-58f4-4eed-af7d-3c2b833471dd)

Reply:
I added another test to hit this available limit.  Keeping the "intel" test the same.